### ISSUE TITLE
Use StackAPI to display php.js-tagged questions?

### ISSUE THREAD

**Issue by brettz9** [kvz/locutus#283]
Just a throwaway idea... You might be able to use the following to display php.js-tagged questions from StackOverflow in place of comments...:

https://api.stackexchange.com/docs

**Comment by kvz** [comment]
Saying it's closed for me. Is that the right link?

**Comment by brettz9** [comment]
I changed the link--does that help? It just shows a button to request access for the beta...

**Comment by kvz** [comment]
While i like the idea of showing that on the site, i think stackshare is unrelated from stack overflow? see http://stackshare.io/about

**Comment by brettz9** [comment]
Whoops--they got me :)  I've updated my OP to add a link to SO's API though

**Comment by kvz** [comment]
@brettz9 I did some work to redo the website over the weekend. Documented bits of it here: 
https://github.com/kvz/phpjs#website and here: http://phpjs.org/blog/2016/04/02/jekyll/.

Would you like to give adding this a try?

The steps would / _should_ be:

```
npm install
npm run website:start # keep an eye on your local browser, it will refresh at the right local page install as you make changes
$EDITOR website/_layouts/function.html
```

I don't have a Windows setup, so I expect some trouble. But in theory it could work as I don't rely on any hard unixy dependency for this I think. We might have to use a few `path.join`s instead of hardcoded `/` here and there. But this might be a good chance for us to make this new setup Windows compatible and give you a chance to hack on the website.

**Comment by brettz9** [comment]
Ok, I can look to give it a try... Just one question though...Looking at https://github.com/kvz/phpjs/blob/master/package.json#L47 , is `npm run version:replace` not going to fail without `version:replace` within your `scripts`?

Also if you want cross-platform, Makefiles aren't very friendly for this unless you require Cygwin or such, and even then, it might not work perfectly.

**Comment by brettz9** [comment]
Also, while we're talking builds...Have you considered moving to the ES6 module import syntax at least for internal dependencies?

**Comment by brettz9** [comment]
And one more question--since you're using Node anyways, aren't there any decent Node platforms for blogging to avoid the need for a Ruby dependency?

**Comment by brettz9** [comment]
You might want to add `npm install` to your blog post to be safe.

`npm install` was successful except for this warning:

> npm WARN eslint-config-standard@5.1.0 requires a peer of eslint-plugin-promise@^1.0.8 but none was installed.

`npm run website:install` failed at `_scripts/postinstall.sh` but after changing it to `cd _scripts && postinstall.sh` the script ran successfully with these warnings:

```
npm WARN optional Skipping failed optional dependency /chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.11
npm WARN optional Skipping failed optional dependency /browser-sync/chokidar/fsevents:
npm WARN notsup Not compatible with your operating system or architecture: fsevents@1.0.11
npm WARN eslint-config-standard@5.1.0 requires a peer of eslint-plugin-promise@^1.0.8 but none was installed.
npm WARN website No description
npm WARN website license should be a valid SPDX license expression
```

However, running `npm run website:start` failed at the use of `rm` in `clean:assets`

Maybe Grunt would be a good way to go for cross-platform support of all of this building?

**Comment by brettz9** [comment]
And I assume the shell script only worked (or appeared to work at least) since I have Cygwin installed..

**Comment by kvz** [comment]
> is npm run version:replace not going to fail without version:replace within your scripts?

Ah yes, fixed now.

> Also if you want cross-platform, Makefiles aren't very friendly for this unless you require Cygwin or such, and even then, it might not work perfectly.

Yeah, it's just a Fakefile though, meaning a transparent wrapper around npm scripts. So you're free to ignore. But convenient for unixy-types. More info here http://kvz.io/blog/2016/02/18/a-universal-makefile-for-javascript/

> Also, while we're talking builds...Have you considered moving to the ES6 module import syntax at least for internal dependencies?

Yes, I've also considered making mode-exports the default for all functions, and maybe rewriting to a regular function, only on the webpage. This would make building our npm easier, and also, make `require('phpjs/sprintf')` straightforward. Seeing as how, even for frontend, folks are using npm via bundling, this seems to make more sense than asking people to just copy-paste global scope functions.

> And one more question--since you're using Node anyways, aren't there any decent Node platforms for blogging to avoid the need for a Ruby dependency?

Yes, we're using Hexo now for Uppy, but I'm afraid we had quite a few issues with it. And most of the docs/support is in Chinese. There's metalsmith, but tbh, the ecosystem of Jekyll is unmatched and with the support of GitHub pages, it's only getting better. Ruby dependency (or adding any other dependency) sucks a bit. But I feel most systems are equiped with Ruby for one reason or another these days.

> You might want to add npm install to your blog post to be safe.

Cool, added

> script ran successfully with these warnings

I see, that sucks. It seems that file watching is not supported on Windows then. This is required for the automatic kicking of jekyll and reloading of your webbrowser. Could you try `bundle exec jekyll serve --watch` inside the `website` directory for me instead? Maybe via Ruby file-watching you can at least still have auto-building, even if no auto browser refreshing.

> And I assume the shell script only worked (or appeared to work at least) since I have Cygwin installed..

I hear you. Not a perfect run, sorry to have built your hopes up Brett :/

But perhaps we can figure out if going straight to Jekyll as mentioned, will work on Windows, and then for the time being just add a Windows specific npm run script for that?

I kinda dislike Grunt :/

**Comment by kvz** [comment]
> And most of the docs/support is in Chinese.

Haha, it just hit me that's not at all a problem for you :smile:

**Comment by brettz9** [comment]
Well, Chinese is *always* a problem for me. :) Should know a lot more for having lived there as long as I have. Sorry, I still mean to get back to this when I get a chance.

**Comment by brettz9** [comment]
I ran `gem install bundle` at a ruby command line, and I was going to do the same for `jekyll`, but I noticed some Windows-specific install instructions at https://jekyllrb.com/docs/windows/#auto-regeneration . You want me to try that and/or what u suggested?

**Comment by kvz** [comment]
That would be cool, although I'm also afraid I'm sending you into a snake pit as I have not tested this flow  myself before. Maybe it's easier I add the stackoverflow idea myself then, for project overhaul I've planned (blogged about it, ICYMI).

I'm thinking this "bash" on Windows 10 will make all of this a lot easier in the near future, so why sink a huge ton of time into this problem when it will disappear in a few months :)

**Comment by kvz** [comment]
php.js is now Locutus (http://locutusjs.io/2016/05/announcing-locutus/)
In the process, I've changed its site to Hexo after all. It was a lot of work but I hope this will encourage Windows contributions since we're all just npm now. There might be still one or two cases in our npm scripts though. Like a `cp` for copy. I did replace `rm -rf` with the `rimraf` package already which offers a cross-platform alternative to it.
